contrib/kl_layer/nnet_distance.h

```diff
@@ -0,0 +1,87 @@
+#ifndef NNET_DISTANCE_H_
+#define NNET_DISTANCE_H_
+
+#include "nnet_activation.h"
+#include "nnet_common.h"
+#include <cmath>
+#include <cstdlib>
+
+namespace nnet {
+
+struct distance_config {
+    // IO size
+    static const unsigned n_in = 10;
+    static const unsigned n_out = 1;
+
+    // Internal data type definitions
+    typedef float accum_t;
+    typedef float sum_t;
+    typedef ap_fixed<18, 8> exp_table_t;
+
+    // Internal info
+    static const unsigned table_size = 1024;
+    static constexpr float exp_range = 1024;
```

Comment:
Does this really need to be float? It would likely have bad QoR if it is not a power-of-two integer.

Reply:
what should it be instead of float?